datafusion/functions-nested/src/map.rs

```diff
@@ -238,7 +241,50 @@ impl ScalarUDFImpl for MapFunc {
     fn invoke(&self, args: &[ColumnarValue]) -> Result<ColumnarValue> {
         make_map_batch(args)
     }
+
+    fn documentation(&self) -> Option<&Documentation> {
+        Some(get_map_doc())
+    }
 }
+
+static DOCUMENTATION: OnceLock<Documentation> = OnceLock::new();
+
+fn get_map_doc() -> &'static Documentation {
+    DOCUMENTATION.get_or_init(|| {
+    Documentation::builder()
+        .with_doc_section(DOC_SECTION_SPECIAL)
```

Comment:
I think I am missing something -- it seems like the `map` function should be in a "Map Functions" section to follow the existing documentation: 

https://datafusion.apache.org/user-guide/sql/scalar_functions.html#map-functions

Reply:
> It seems like `map` is still a scalar function -- I was thinking that the `DocSection` was what was special (as in it would be `DOC_SECTION_SPECIAL` that was part of `scalar_doc_sections` 🤔
I think you mentioned above that you wanted the function to put in a doc_section_special here. So should I make the map function go under 'doc section map' instead?

Reply:
I would say so, yes. My original thought was that unnest and make_map would be the two special functions but digging through the planner code it seems that make_map just calls into map udf  (https://github.com/apache/datafusion/issues/11434) so I think the only special one is unnest.

Reply:
Alright, so the direction of this pr now is to just add a static page with unnest?

Reply:
Sorry for the confusion -- I think there are two distinct action items:
1. Migrate the documentation for `map` and related functions to a "Map Functions" section
2. Create a new static page for "special functions" (different than Scalar Functions) where we can put `unnest`


Perhaps we can turn this PR into the `map` and related functions in a "Map Functions" section and make another PR to move the `unnest` docs to the new page.

I am happy to help with either

Thanks again for your help @jonathanc-n and @Omega359